src/authorizations/utils/permissions.ts

```diff
@@ -269,3 +270,38 @@ export const formatApiPermissions = (permissions, orgID) => {
   })
   return apiPerms
 }
+
+export const generateDescription = apiPermissions => {
+  let generatedDescription = ''
+
+  if (apiPermissions.length > 2) {
+    const actions = []
+    apiPermissions.forEach(perm => {
+      actions.push(perm.action)
+    })
+    const isRead = actions.some(action => action === 'read')
+    const isWrite = actions.some(action => action === 'write')
+
+    if (isRead && isWrite) {
+      generatedDescription += `Read Multiple Write Multiple`
+    } else if (isRead) {
+      generatedDescription += `Read Multiple`
+    } else if (isWrite) {
+      generatedDescription += `Write Multiple`
+    }
+  } else {
+    apiPermissions.forEach(perm => {
+      if (perm.resource.name) {
+        generatedDescription += ` ${capitalize(perm.action)} ${
+          perm.resource.type
+        } ${perm.resource.name} `
+      } else {
+        generatedDescription += ` ${capitalize(perm.action)} ${
+          perm.resource.type
+        } `
+      }
+    })
+  }
+
+  return generatedDescription
+}
```

Comment:
let's see a bunch of unit tests for this method!

I'd like you to think about all the different permutations that this can produce and try to write a test case for each type of one. the more cases, the more likely you are to find edge cases and head off any weirdness before users see it.

Let me know if you'd like to pair up on this!